src/gfn/gym/bitSequence.py

```diff
@@ -0,0 +1,755 @@
+
+        Args:
+            states: A batch of states.
+            actions: A batch of actions.
+
+        Returns:
+            torch.Tensor: A batch of next states.
+        """
+        is_exit = actions.is_exit
+        old_tensor = states.tensor.clone()
+        old_tensor[~is_exit, states.length] = actions.tensor[~is_exit].squeeze()
+        old_tensor[is_exit] = torch.full_like(
+            old_tensor[is_exit],
+            self.n_actions - 1,
+            dtype=torch.long,
+            device=old_tensor.device,
+        )
+        return old_tensor
+
+    def backward_step(self, states: BitSequenceStates, actions: Actions) -> torch.Tensor:
+        """Function that takes a batch of states and actions and returns a batch of previous
+        states. Does not need to check whether the actions are valid or the states are sink states.
+        Operates on flattened states only.
+
+        Args:
+            states: A batch of states.
+            actions: A batch of actions.
+
+        Returns:
+            torch.Tensor: A batch of previous states.
+        """
+        assert (
+            actions.tensor.squeeze()
+            == states.tensor[
+                torch.arange(states.tensor.shape[0]),
+                states.length - 1,
+            ]
+        ).all()
+        old_tensor = states.tensor.clone()
+        old_tensor[..., states.length - 1] = -1
+        return old_tensor
+
+    def _step(self, states: BitSequenceStates, actions: Actions):
+        """
+        Perform a step in the environment by applying the given actions to the current states.
+        Args:
+            states (BitSequenceStates): The current states of the environment.
+            actions (Actions): The actions to be applied to the current states.
+        Returns:
+            BitSequenceStates: The new states of the environment after applying the actions.
+        """
+
+        new_states = super(DiscreteEnv, self)._step(states, actions)
+        assert isinstance(new_states, BitSequenceStates)
+        new_states.length = states.length + 1
+
+        self.update_masks(new_states)
+
+        return new_states
+
+    def _backward_step(self, states: BitSequenceStates, actions: Actions):
+        """
+        Perform a backward step in the environment by undoing the given actions to the current states.
+
+        Args:
+            states (BitSequenceStates): The current states of the environment.
+            actions (Actions): The actions to be applied to the current states.
+
+        Returns:
+            BitSequenceStates: The new states after performing the backward step.
+        """
+        new_states = super(DiscreteEnv, self)._backward_step(states, actions)
+        assert isinstance(new_states, BitSequenceStates)
+        new_states.length = states.length - 1
+        self.update_masks(new_states)
+        return new_states
+
+    def make_modes_set(self, seed: int = 42) -> torch.Tensor:
```

Comment:
Could you explain in one or two sentences in the docstring how the modes are defined from H?